zerver/views/realm.py

```diff
@@ -104,5 +107,15 @@ def update_realm(request, user_profile, name=REQ(validator=check_string, default
             do_set_realm_notifications_stream(realm, new_notifications_stream,
                                               notifications_stream_id)
             data['notifications_stream_id'] = notifications_stream_id
+    if signup_notifications_stream_id is not None:
+        if realm.signup_notifications_stream is None or \
+           realm.signup_notifications_stream.id != signup_notifications_stream_id:
+            new_signup_notifications_stream = None
+            if signup_notifications_stream_id >= 0:
+                (new_signup_notifications_stream, recipient, sub) = access_stream_by_id(user_profile,
```

Comment:
I think it would be good to follow the formatting of the notifications_stream_id here:
```
access_stream_by_id(
     user_profile, signup_...
```